lint/src/main/kotlin/io/homeassistant/lint/serialization/MissingSerializableAnnotationIssue.kt

```diff
@@ -0,0 +1,189 @@
+package io.homeassistant.lint.serialization
+
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+import com.intellij.psi.PsiClassType
+import com.intellij.psi.PsiType
+import com.intellij.psi.PsiWildcardType
+import io.homeassistant.lint.utils.Logger
+import kotlin.reflect.jvm.jvmName
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.UElement
+import org.jetbrains.uast.UExpression
+
+private const val KOTLINX_JSON_CLASS = "kotlinx.serialization.json.Json"
+private const val KOTLINX_SERIALIZABLE_ANNOTATION = "kotlinx.serialization.Serializable"
+private const val METHOD_NAME = "encodeToString" // TODO handle decodeFromString as well
```

Comment:
TODO in this PR?

Reply:
Well that's up to you ^^ but I found the code already a bit hard so I told myself that it might be better to do it in a separate PR. But if you feel it would be ok I can do it here too.

Reply:
Let's not do it here then, if it is another PR that should be smaller, easier to review and also a nice example when adding more lint rules in the future.